pdp/ast/jast/obligations.go

```diff
@@ -24,11 +25,43 @@ func (ctx context) unmarshalObligationItem(d *json.Decoder) (pdp.AttributeAssign
 			return newUnknownAttributeError(k)
 		}
 
-		e, err = ctx.unmarshalValueByType(a.GetType(), d)
+		delim, val, err := jparser.CheckObjectStartOrValue(d, "argument")
 		if err != nil {
-			return bindError(err, k)
+			return err
+		}
+		if delim == "" {
+			e, err = ctx.unmarshalValueByTypeObject(a.GetType(), val)
+			if err != nil {
+				return err
+			}
+		} else if delim == jparser.DelimObjectStart {
+			e, err = ctx.unmarshalExpression(d)
+			if err != nil {
+				return bindError(err, k)
+			}
+		} else if delim == jparser.DelimArrayStart {
+			val, err := jparser.GetArray(d, "obligations")
+			e, err = ctx.unmarshalValueByTypeObject(a.GetType(), val)
+			if err != nil {
+				return err
+			}
+			/*
```

Comment:
No need to comment code - just remove it. If you need it for later make other branch.

Reply:
My bad. It's no longer needed. I'll remove it.

Reply:
Done